libs/datamodel/core/src/transform/helpers/value_validator.rs

```diff
@@ -180,6 +180,14 @@ impl ValueValidator {
         }
     }
 
+    /// Checks if the wrapped value is an array
+    pub fn is_array(&self) -> bool {
+        match self.value {
+            ast::Expression::Array(_, _) => true,
+            _ => false,
+        }
+    }
```

Comment:
I think an `ast::Expression::is_array()` method would be more general, and not less ergonomic here, since you could write `validator.value.is_array()`

Reply:
oh yes, good point! I've changed that now.